examples/ogbn_train.py

```diff
@@ -0,0 +1,321 @@
+            if i != self.num_layers - 1:
+                x = F.elu(x)
+                x = F.dropout(x, p=self.dropout, training=self.training)
+        return x
+
+    def inference(self, x_all: Tensor) -> Tensor:
+        pbar = tqdm(total=x_all.size(0) * self.num_layers)
+        pbar.set_description('Evaluating')
+
+        # Compute representations of nodes layer by layer, using *all*
+        # available edges. This leads to faster computation in contrast to
+        # immediately computing the final representations of each batch.
+        for i in range(self.num_layers):
+            xs = []
+            for batch in subgraph_loader:
+                x = x_all[batch.n_id].to(device)
+                edge_index = batch.edge_index.to(device)
+                x = self.convs[i](x, edge_index) + self.skips[i](x)
+                x = x[:batch.batch_size]
+                if i != self.num_layers - 1:
+                    x = F.elu(x)
+                xs.append(x.cpu())
+                pbar.update(batch.batch_size)
+
+            x_all = torch.cat(xs, dim=0)
+
+        pbar.close()
+        return x_all
+
+
+def train(epoch: int) -> Tuple[Tensor, float]:
+    model.train()
+
+    pbar = tqdm(total=split_idx['train'].size(0))
+    pbar.set_description(f'Epoch {epoch:02d}')
+
+    total_loss = total_correct = 0
+    for batch in train_loader:
+        optimizer.zero_grad()
+        out = model(batch.x, batch.edge_index.to(device))[:batch.batch_size]
+        y = batch.y[:batch.batch_size].squeeze().to(torch.long)
+        loss = F.cross_entropy(out, y)
+        loss.backward()
+        optimizer.step()
+
+        total_loss += float(loss)
+        total_correct += int(out.argmax(dim=-1).eq(y).sum())
+        pbar.update(batch.batch_size)
+
+    pbar.close()
+    loss = total_loss / len(train_loader)
+    approx_acc = total_correct / split_idx['train'].size(0)
+    return loss, approx_acc
+
+
+@torch.no_grad()
+def test(loader: NeighborLoader, val_steps=None) -> float:
+    model.eval()
+
+    total_correct = total_examples = 0
+    for i, batch in enumerate(loader):
+        if val_steps is not None and i >= val_steps:
+            break
+        batch = batch.to(device)
+        batch_size = batch.num_sampled_nodes[0]
+        out = model(batch.x, batch.edge_index)[:batch_size]
+        pred = out.argmax(dim=-1)
+        y = batch.y[:batch_size].view(-1).to(torch.long)
+
+        total_correct += int((pred == y).sum())
+        total_examples += y.size(0)
+
+    return total_correct / total_examples
+
+
+if args.use_gat:
+    model = GAT(
+        dataset.num_features,
+        num_hidden_channels,
+        dataset.num_classes,
+        num_layers=num_layers,
+        dropout=args.dropout,
+        heads=args.num_heads,
+    )
+else:
+    model = SAGE(
+        dataset.num_features,
+        num_hidden_channels,
+        dataset.num_classes,
+        num_layers=num_layers,
+        dropout=args.dropout,
+    )
```

Comment:
Mind running `SAGE` case as it was never run before?

Reply:
Default model used for training is GraphSage model.